pkg/query/remote_engine.go

```diff
@@ -289,8 +306,15 @@ func (r *remoteQuery) Exec(ctx context.Context) *promql.Result {
 				warnings.Add(errors.New(warn))
 				continue
 			}
+			if s := msg.GetStats(); s != nil {
+				qryStats = s
+				continue
+			}
 
 			ts := msg.GetTimeseries()
+			if ts == nil {
```

Comment:
Why do we need this check? At this point it has to be a timeseries right?

Reply:
We dont have a linter for exhaustive check, so if someone adds a new type to the `oneof` in the proto, it might cause problems. I will remove for now.